### PR TITLE
String view unordered map search

### PR DESCRIPTION
**Pull request by biodranik**

### CODE REVIEW

**Comment by biodranik** [comment]
It avoids unnecessary memory allocation and demonstrates how to do it properly in other cases.

**Review by vng — COMMENTED** [review]
Well, I'd rather wait until std::hash will be is_transparent (like it should! be) ..
Anyway LGTM

File: indexer/brands_holder.hpp [hunk 1]
```diff
@@ -57,7 +57,17 @@ class BrandsHolder
...
-  std::unordered_map<std::string, std::shared_ptr<Brand>> m_keyToName;
+  struct StringViewHash
+  {
+    using hash_type = std::hash<std::string_view>;
+    using is_transparent = void;
+
+    // std::size_t operator()(const char* str) const        { return hash_type{}(str); }
+    std::size_t operator()(std::string_view str) const   { return hash_type{}(str); }
+    std::size_t operator()(std::string const & str) const { return hash_type{}(str); }
+  };
```
Comment (vng) [inline indexer/brands_holder.hpp:68]: Please, use size_t without std

File: indexer/brands_holder.hpp [hunk 2]
```diff
@@ -57,7 +57,17 @@ class BrandsHolder
...
 
-  std::unordered_map<std::string, std::shared_ptr<Brand>> m_keyToName;
+  struct StringViewHash
+  {
+    using hash_type = std::hash<std::string_view>;
+    using is_transparent = void;
+
+    // std::size_t operator()(const char* str) const        { return hash_type{}(str); }
+    std::size_t operator()(std::string_view str) const   { return hash_type{}(str); }
+    std::size_t operator()(std::string const & str) const { return hash_type{}(str); }
```
Comment (biodranik) [inline indexer/brands_holder.hpp:67]:
```suggestion
    size_t operator()(std::string_view str) const   { return hash_type{}(str); }
    size_t operator()(std::string const & str) const { return hash_type{}(str); }
```

**Comment by biodranik** [comment]
> Well, I'd rather wait until std::hash will be is_transparent (like it should! be) ..

Wait till C++2099?

**Review by vng — APPROVED** [review]

**Pull request merged**